esx_service/vmdk_ops.py

```diff
@@ -1075,22 +1084,101 @@ def load_vmci():
    else:
        lib = CDLL(os.path.join(LIB_LOC, "libvmci_srv.so"), use_errno=True)
 
+def getLock(lockname):
+    '''
+    Create or return a existing lock identified by lockname.
+    '''
+    global rsrcLocks
+
+    with managerLock:
+        try:
+            lock = rsrcLocks[lockname]
+            logging.debug("getLock(): return existing lock %s", lockname)
+        except KeyError as e:
+            lock = threading.Lock()
+            rsrcLocks[lockname] = lock
+            logging.debug("getLock(): return new lock %s", lockname)
+    return lock
+
+def execRequestThread(client_socket, cartel, request):
+    '''
+    Execute requests in a thread context with a per volume locking. 
+    '''
+    # Before we start, block to allow main thread or other running threads to advance.
+    # https://docs.python.org/2/faq/library.html#none-of-my-threads-seem-to-run-why
+    time.sleep(0.001)
+
+    # Get VM name & ID from VSI (we only get cartelID from vmci, need to convert)
+    vmm_leader = vsi.get("/userworld/cartel/%s/vmmLeader" % str(cartel))
+    group_info = vsi.get("/vm/%s/vmmGroupInfo" % vmm_leader)
+    vm_name = group_info["displayName"]
+    cfg_path = group_info["cfgPath"]
+    uuid = group_info["uuid"]
```

Comment:
Not related to this change, but we hold no reference to this VM and that can exit any time and the VSI entries may be invalid. Better to have it in a try except block, no way that we can return any error in case these accesses return error.